### PR TITLE
add keyboard aware support to FlatList

### PR DESCRIPTION
**Pull request by chrisy-lili**

### CODE REVIEW

File: lib/KeyboardAwareFlatList.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  componentWillMount: function () {
+    this.setViewIsInsideTabBar(this.props.viewIsInsideTabBar)
+    this.setResetScrollToCoords(this.props.resetScrollToCoords)
+  },
+
+  render: function () {
+    return (
+      <FlatList
+        ref='_rnkasv_keyboardView'
```
Comment (alvaromb) [inline lib/KeyboardAwareFlatList.js]: We should use callback refs instead of string refs.

Comment (chrisy-lili) [inline lib/KeyboardAwareFlatList.js]: strings are used in the existing KeyboardAwareScrollView and KeyboardAwareListView components. KeyboardAwareMixin also references ref as `this.refs._rnkasv_keyboardView`, which I think should be `this._rnkasv_keyboardView` for callback refs. Do you want me to update everywhere to use callback refs in this PR? Or, leave it as is, and open another PR to switch to callback refs?

**Comment by Swordsman-Inaction** [comment]
@chrisy-lili Would you like to update this PR for Android Support? Check `KeyboardAwareScrollView`'s source code, I just add some codes for Android Support.

File: lib/KeyboardAwareFlatList.js (hunk 1 above)
Comment (Swordsman-Inaction) [inline lib/KeyboardAwareFlatList.js]: @alvaromb How do you think about @chrisy-lili 's suggestion? We could have another PR for refs change.

**Comment by chrisy-lili** [comment]
@Swordsman-Inaction I've updated the PR with android support based on your changes in the other component. Thanks for pointing me to that update. This PR is ready for review again :)

**Comment by Swordsman-Inaction** [comment]
@alvaromb Can you review this again? I think we can change string refs later.

It's ready to release a new version after merging this PR.

**Comment by alvaromb** [comment]
@Swordsman-Inaction done!

Are we ready to release?

**Comment by Swordsman-Inaction** [comment]
@alvaromb  Yes! 👏

**Comment by quadsurf** [comment]
what's the status on this? pls also ensure support for when the FlatList has the '''inverted''' prop enabled (expected behavior is that the scroll event will travel in the reverse or opposite direction)

this would be a very common use case for chat-based apps

**Comment by alvaromb** [comment]
It's ready, but I'm still changing some things before releasing it. See this: https://github.com/APSL/react-native-keyboard-aware-scroll-view/issues/163

**Comment by Naiba01** [comment]
Is there an example to show how to use KeyboardAwareFlatList? It always reponse that this.refs._rnkasv_keyboardView.getScrollResponder is not a function.

**Comment by alvaromb** [comment]
It's `this. _rnkasv_keyboardView`, remove `refs`.

**Pull request merged**